core/mchat.php

```diff
@@ -1056,7 +1056,13 @@ public function assign_messages($rows, $page = '')
 
 			$message_age = time() - $row['message_time'];
 			$minutes_ago = $this->get_minutes_ago($message_age);
-			$datetime = $this->user->format_date($row['message_time'], $this->settings->cfg('mchat_date'), true);
+			// If relative time is selected, also display "today" / "yesterday", else display absolute time.
+			if ($this->settings->cfg('mchat_relative_time'))
+			{
+				$datetime = $this->user->format_date($row['message_time'], $this->settings->cfg('mchat_date'), false);
```

Comment:
This won't play nicely with the [Relative Dates extension](https://www.phpbb.com/customise/db/extension/relative_dates/). If it is enabled and the third parameter is `false` the HTML markup is broken. I think in the `if` condition above there needs to be an additional check along the lines of `&& !relative_dates_extension_enabled`.